src/App.tsx

```diff
@@ -1,26 +1,125 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [fetchedTodos, setFetchedTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterBy, setFilterBy] = useState<'all' | 'active' | 'completed'>(
+    'all',
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [completedTodoIds, setCompletedTodoIds] = useState<number[] | null>(
```

Comment:
There might be completed todos, and it's not these todos - i.e. this name is misleading
use `todoIdsToDelete`

use empty array as initial value - that's common way to do it for arrays in state